### PR TITLE
Added documentation for Lucky numbers

### LINKED ISSUE #265: Documentation for Lucky number
## 📚 Documentation

Documentation for Lucky number.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by nidh-ish**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added documentation for Lucky numbers.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #265

### CODE REVIEW

File: docs/mathematics/lucky-numbers.mdx [hunk 1]
```diff
@@ -0,0 +1,336 @@
+---
+title: Lucky Number
+slug: /mathematics/lucky-numbers
+---
+
+## Introduction
+
+Lucky numbers are a subset of integers that are generated by the result of a sieving process. The sieving process eliminates numbers based on their position in the remaining set.
```
Comment (HarshCasper) [inline docs/mathematics/lucky-numbers.mdx:8]: Can you give a slightly better introduction here?

File: docs/mathematics/lucky-numbers.mdx [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+
+- The next surviving number is nine. Delete every ninth number from the above set.
+  1, 3, 7, 9, 13, 15, 21, 25, 31, . . .
+
+- Similarly, the next surviving number is thirteen. The above process is continued.
+
+- The number which doesn't get deleted due to the above process is a LUCKY NUMBER.
+
+- First few lucky numbers are: 1, 3, 7, 9, 13, 15, 21, 25, . . .  
+
```
Comment (HarshCasper) [inline docs/mathematics/lucky-numbers.mdx:35]: A better and more consumable example would be appreciated. Start with example first and then proceed with explanation

File: docs/mathematics/lucky-numbers.mdx [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+    
+    2) `count`: to store the index of the remaining numbers in `array`, used for further elimination.
+    
+    3) `check`: to check the condition for termination.
+  - Initially, both `array` and `count` contain integers from `1` to `n`.
+  - Now we remove all the even numbers from the `array`, i.e., fill all the even places with `-1`.
+  - Since we didn't actually remove the values from the `array`, just filled them with `-1`, we use `count` to store the updated index of remaining numbers in the array.
+  - Now we iterate till every possible elimination is done from the `array`. To terminate, we keep a check of the number of `-1's` in `array` and store them in `check` at the end of every iteration.
+  - If we get the same count of `-1` at two consecutive indices of `check`, we know for sure that all possible values have been eliminated from the `array` and we terminate.
+  - Finally print the values of `array` except `-1`.
```
Comment (HarshCasper) [inline docs/mathematics/lucky-numbers.mdx:53]: Use crisp points. And use sub-bullets instead of ordered ones.

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by nidh-ish** [comment]
@HarshCasper Please have a look.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**